python/tvm/script/parser.py

```diff
@@ -1138,6 +1141,29 @@ def transform_TypeTuple(self, node):
         """
         return [self.transform(value) for value in node.values]
 
+    def transform_TypeApply(self, node):
+        """Visitor for Type[Type] expressions.
+
+        Mostly used for ``T.Ptr`` expressions.
+        """
+        func = self.transform(node.func_name)
+
+        if not isinstance(func, ty.TypeGeneric):
```

Comment:
The `TypeApply` functionality also occurs with the `T.Tuple[type1,type2]` type annotation.  While I couldn't find any usage of `T.Tuple` type annotations, `GenericTupleType` exists in `tvm.script.tir.ty` and would also use `TypeApply` in the synr ast.  I've added a check for `hasattr(func, __getitem__)`, since that expresses the intent that this is a type that can accept type arguments, and would work for both `T.Ptr` and `T.Tuple`.